lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -32,384 +57,165 @@ def get(section: str, option: str, fallback=None) -> str:
+            (0, 586, ["md5.1", "md5.2"]),
+        ],
     )
-    def test_tarball_size(
-        self, monkeypatch, make_logger, caplog, min_size, max_size, total
+    def test_buckets(
+        self, monkeypatch, make_logger, reset, min_size, max_size, targets
     ):
-        """Test for the valid size and path of the tarball. We are intentionally
-        causing Path.resolve() to fail as a means to short-circuit the CUT
-        while still enabling the testing of the size filtering."""
-
-        def mock_resolve(self: Path, strict: bool = False):
-            """
-            Mock replacement for Path.resolve()
-
-            Args:
-                strict: Boolean value to make sure path exist if True.
-
-            Returns:
-                Raising FileNotFoundError Exception
-            """
-            assert strict
-            raise FileNotFoundError(f"No such file or directory: '{self}'")
-
-        monkeypatch.setattr(Path, "glob", lambda path, args: self.tarlist)
-        monkeypatch.setattr(Path, "resolve", mock_resolve)
-        monkeypatch.setattr(Path, "stat", self.mock_stat)
-        obj = UnpackTarballs(MockConfig, make_logger)
-        result = obj.unpack_tarballs(min_size, max_size)
-        if result.total > 0:
-            for tar in self.tarlist:
-                assert (
-                    f"Tarball link, '{tar}', does not resolve to a real location: No such file or directory: '{tar}'"
-                    in caplog.text
-                )
-        assert result.total == total
-        assert result.success == 0
-
-    def test_unpack_md5_error(self, monkeypatch, make_logger, caplog):
-        """Show that when md5 file path cannot be accessed it throws a
-        FileNotFoundError Exception"""
-
-        mocks_called = []
-
-        def mock_md5sum(filepath: Path):
-            """
-            Mock of md5sum function
-
-            Args:
-                filepath: Tarball file path
-
-            Returns:
-                Raises FileNotFoundError Exception
-            """
-            mocks_called.append("mock_tarball_md5")
-            raise FileNotFoundError(f"No such file or directory: '{filepath}.md5'")
-
-        monkeypatch.setattr("pbench.server.utils.md5sum", mock_md5sum)
-        obj = UnpackTarballs(MockConfig, make_logger)
-        with pytest.raises(FileNotFoundError):
-            obj.unpack(self.tar, None)
-            assert (
-                f"Getting md5 value of tarball '{self.tar}' failed: No such file or directory: '{self.tar}.md5'"
-                in caplog.text
+        """Test that the tarbal list is filtered by the unpack bucket size
+        configuration."""
+
+        with monkeypatch.context() as m:
+            m.setattr("pbench.server.unpack_tarballs.Path", MockPath)
+            m.setattr("pbench.server.unpack_tarballs.Sync", MockSync)
+            m.setattr("pbench.server.unpack_tarballs.CacheManager", MockCacheManager)
+            m.setattr(Metadata, "getvalue", getvalue)
```

Comment:
We shouldn't need the `with` statement:  for three of the mocks, there shouldn't be any conflicts with PyTest or the other test superstructure; for `Path`, I _think_ that that patch point is specific enough (i.e., it's mocking the `Path` references only in the `unpack_tarballs` module) that it won't cause any conflicts, either.  So, just using the `monkeypatch` fixture directly should suffice (it will revoke the mocks at the end of the function, even without the `with`).

Same for the other tests, here, I believe.